.github/workflows/pr-checks.yml

```
@@ -13,12 +13,34 @@ jobs:
     - name: Checkout the source code
       uses: actions/checkout@v3
 
-    - name: Install cargo contract
+    - name: Install & display rust toolchain
       run: |
+        rustup show
         rustup component add rust-src
+
+    - name: Check targets are installed correctly
+      run: rustup target list --installed
+
+    - name: Cache Crates
+      uses: actions/cache@v3
+      with:
+        path: ~/.cargo
+        key: ${{ runner.os }}-rust-${{ hashFiles('rust-toolchain.toml') }}
+        restore-keys: |
+          ${{ runner.os }}-rust
+
+    - name: Check if cargo-contract exists
+      id: check-cargo-contract
+      continue-on-error: true
+      run: cargo contract --version
+
+    - name: Install cargo contract
+      if: ${{ steps.check-cargo-contract.outcome == 'failure' }}
+      run: |
         sudo apt-get install binaryen
         cargo install cargo-dylint dylint-link
-        cargo install --force --locked cargo-contract
+        # Until 2.0.0 official release, we need to specify --version expressly
+        cargo install --force --locked cargo-contract --version 2.0.0
```

```suggestion
        cargo install --force --locked cargo-contract
```
v2 is officially released. No need to specify `--version`

---

Sure!